### ISSUE TITLE
[Bug] Crash on pressing search widget, Home, then search widget

### ISSUE THREAD

**Issue by colintheshots** [mozilla-mobile/fenix#4779]
## Steps to reproduce
1. Install the Fenix search widget on a home screen.
2. Press the search widget icon or text.
3. Minimize the app.
4. Press the search widget icon or text.

### Expected behavior
The app loads the search fragment again.

### Actual behavior
The app crashes with this stack trace:
```
2019-08-16 15:35:19.492 org.mozilla.fenix.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix.debug, PID: 24659
    java.lang.IllegalStateException: Fragment SearchFragment{833960a (31d40128-3976-46d7-ad27-2cc72dbedb39)} not attached to a context.
        at androidx.fragment.app.Fragment.requireContext(Fragment.java:803)
        at androidx.fragment.app.Fragment.getResources(Fragment.java:867)
        at org.mozilla.fenix.search.SearchFragment.updateSearchEngineIcon(SearchFragment.kt:241)
        at org.mozilla.fenix.search.SearchFragment.access$updateSearchEngineIcon(SearchFragment.kt:48)
        at org.mozilla.fenix.search.SearchFragment$onViewCreated$5.invoke(SearchFragment.kt:193)
        at org.mozilla.fenix.search.SearchFragment$onViewCreated$5.invoke(SearchFragment.kt:48)
        at mozilla.components.lib.state.ext.FragmentKt$consumeFrom$1.invokeSuspend(Fragment.kt:37)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:238)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```
### Device information

* Android device: Pixel 3XL, Android Q
* Fenix version: Latest master, 8/16/2019

**Comment by ekager** [comment]
This specific issue seems to be fixed in the newest AC snapshot (related to consumeFrom)

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest Nightly build 1.0.1934 from 8/23, with Xiaomi Mi 8 Lite (Android 9), Samsung Galaxy S8 (Android 9), Pixel XL (Android Q Beta 4), Google Pixel 3a (Android 9), Huawei Mate 20 Lite (Android 8.1.0), and Samsung Galaxy S7 (Android 7).
